ODIN_II/SRC/adders.cpp

```diff
@@ -1311,12 +1315,12 @@ void reallocate_pins(nnode_t *node, nnode_t *next_node)
  *-------------------------------------------------------------------------*/
 void free_op_nodes(nnode_t *node)
 {
-	int i;
-	for (i = 0; i < node->num_output_pins; i++)
+	for (int i = 0; i < node->num_output_pins; i++)
 	{
 		if (node->output_pins[i]->net != NULL)
```

Comment:
should we also free the pin and NULL them ?

Reply:
after looking at free_nnode i see that it rolls it's own free_npin, 

https://github.com/verilog-to-routing/vtr-verilog-to-routing/blob/1ab2b938b8e70b10347045f9fee83f7660ca45e6/ODIN_II/SRC/netlist_utils.cpp#L103-L118

can you make it call free_npin there

other than that looks good to me!

Reply:
The pins can't be freed here, because some of them get added to another node in the match_node function. That's why I was freeing the ones that don't get connected in the match_node function. Freeing the pins in free_op_node would cause the program to segfault

Reply:
gotcha! makes sense then.